client/src/components/Header/HeaderMain/CategoryMenu/CategoryMenu.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import menuIcon from './menuIcon.png';
+import CategoryLayer from './CategoryLayer';
+
+const Wrapper = styled.div`
+  position: relative;
+
+  .button {
+    display: flex;
+    align-items: center;
+    cursor: pointer;
+
+    .icon {
+      width: 20px;
+      height: auto;
+
+      margin-right: 8px;
+    }
+
+    .text {
+      font-size: 14px;
+      color: #333;
+    }
+  }
+`;
+
+const CategoryMenu = (): JSX.Element => {
+  const [showLayer, setShowLayer] = useState(false);
+  const handleLayerButtonClick = useCallback(
+    () => setShowLayer(!showLayer),
+    [showLayer, setShowLayer]
+  );
```

Comment:
setState 같은 경우에 이전 값을 사용하면 함수로 업데이트 시키는 것도 좋을 것 같은데 어떻게 생각하시나요? 그렇게 하면 showLayer를 dep 배열에서 제거할수도 있을 것 같네요 :) 

```suggestion
  const handleLayerButtonClick = useCallback(
    () => setShowLayer((prevShowLayer) => !prevShowLayer),
    [setShowLayer]
  );
```

Reply:
이것도 잘 몰랐던 부분이네요. 고맙습니다. :)